src/fn_call.rs

```diff
@@ -152,8 +159,151 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
+                // First, we extract the vtable pointer from our fat pointer,
+                // and check its alignment
+
+                let vtable_ptr = payload_dyn.to_meta()?.expect("Expected fat pointer!").to_ptr()?; 
+                let data_ptr = payload_dyn.to_scalar_ptr()?;
+                this.memory().check_align(vtable_ptr.into(), this.tcx.data_layout.pointer_align.abi)?;
+
+                // Now, we dereference the vtable pointer.
+                let alloc = this.memory().get(vtable_ptr.alloc_id)?;
+
+                // Finally, we extract the pointer to 'box_me_up'.
+                // The vtable is layed out in memory like this:
+                //
+                //```
+                // <drop_ptr> (usize)
+                // <size> (usize)
+                // <align> (usize)
+                // <method_ptr_1> (usize)
+                // <method_ptr_2> (usize)
+                // ...
+                // <method_ptr_n> (usize)
+                //```
+                //
+                // Since box_me_up is the first method pointer
+                // in the vtable, we use an offset of 3 pointer sizes
+                // (skipping over <drop_ptr>, <size>, and <align>)
+
+                let box_me_up_ptr = alloc.read_ptr_sized(
+                    this,
+                    vtable_ptr.offset(this.pointer_size() * 3, this)?
+                )?.to_ptr()?;
+
+                // Get the actual function instance
+                let box_me_up_fn = this.memory().get_fn(box_me_up_ptr)?;
+                let box_me_up_mir = this.load_mir(box_me_up_fn.def)?;
+
+                // Extract the signature
+                // We know that there are no HRBTs here, so it's fine to use
+                // skip_binder
+                let fn_sig_temp = box_me_up_fn.ty(*this.tcx).fn_sig(*this.tcx);
+                let fn_sig = fn_sig_temp.skip_binder();
+
+                // This is the layout of '*mut (dyn Any + Send)', which
+                // is the return type of 'box_me_up'
+                let dyn_ptr_layout = this.layout_of(fn_sig.output())?;
+
+                // We allocate space to store the return value of box_me_up:
+                // '*mut (dyn Any + Send)', which is a fat 
+                
+                let temp_ptr = this.allocate(dyn_ptr_layout, MiriMemoryKind::UnwindHelper.into());
+                this.machine.box_me_up_tmp_ptr = Some(temp_ptr.clone());
+
+                this.push_stack_frame(
+                    box_me_up_fn,
+                    box_me_up_mir.span,
+                    box_me_up_mir,
+                    Some(temp_ptr.into()),
+                    // We know that this frame will always be unwound,
+                    // since we enable unwinding as soon as we encounter it
+                    // in Evaluator::stack_pop
+                    StackPopCleanup::Goto { ret: None, unwind }
+                )?;
+
+                this.frame_mut().extra.is_box_me_frame = true;
```

Comment:
Please add a comment saying that the code you wrote here basically "goes on" in `pop_frame`, which triggersspecial behavior based on this boolean flag.